### PR TITLE
Add formatting using rustfmt

### LINKED ISSUE #157: Check how to apply style guides
Currently we are using different IDEs (e.g. IntelliJ and VisualCode) which have a different default formatting. It would be nice to be able to apply a standard so that both autoformat it to the same code.

We might consider adding linting for this too to our build pipeline.

### PR DESCRIPTION
**Pull request by Uxio0**
- Code formatting will be checked before compilation on _travis_. If code is not formatted, _travis_ will stop there.
- A PR template with the code formatting requisite has been added.
- README was updated with the `rustfmt` instructions.
- Code can be formatted using `cargo fmt --all` or configuring it in the IDEs you are using https://github.com/rust-lang/rustfmt#running-rustfmt-from-your-editor
- All the code has been formatted, so this commit should be merged asap.
- Closes #157

### CODE REVIEW

**Comment by Uxio0** [comment]
Coverage decreased just reformatting... I guess the coverage tool takes every line as an instruction...

File: src/models/converters/transactions/tests/mod.rs [hunk 1]
```diff
@@ -1,8 +1,8 @@
...
 mod data_size_calculation;
-mod summary;
 mod details;
-pub(super) mod transaction_types;
 pub(super) mod map_status;
 pub(super) mod missing_signers;
-pub(super) mod check_sender_or_receiver;
-pub(super) mod transfer_type_checks;
\ No newline at end of file
+mod summary;
```
Comment (rmeissner) [inline src/models/converters/transactions/tests/mod.rs:6]: Wow this is weird, that the mod visibility has no impact on sorting ... would look nicer if they did :P

**Review by rmeissner — APPROVED** [review]
Having a pre-commit hook would probably be nice :)

**Comment by Uxio0** [comment]
> Having a pre-commit hook would probably be nice :)

I added one script and updated the Readme, but as you know everyone will have to install it. It seems to be working well for me

**Review by rmeissner — APPROVED** [review]

**Pull request merged**